pkg/controller/sidecarterminator/sidecar_terminator_controller.go

```diff
@@ -136,22 +134,25 @@ func (r *ReconcileSidecarTerminator) doReconcile(pod *corev1.Pod) (reconcile.Res
 	if !isInterestingPod(pod) {
 		return reconcile.Result{}, nil
 	}
-
-	sidecarNeedToExecuteKillContainer, sidecarNeedToExecuteInPlaceUpdate, err := r.groupSidecars(pod)
-
+	vk, err := IsPodRunningOnVirtualKubelet(pod, r.Client)
 	if err != nil {
+		klog.ErrorS(err, "SidecarTerminator -- Error occurred when try to check if pod is running on virtual-kubelet", "pod", klog.KObj(pod))
 		return reconcile.Result{}, err
 	}
-
-	if err := r.executeInPlaceUpdateAction(pod, sidecarNeedToExecuteInPlaceUpdate); err != nil {
+	normalSidecarNames, ignoreExitCodeSidecarNames, inplaceUpdateSidecarNames := getSidecarContainerNames(pod, vk)
+	if err = r.executeInPlaceUpdateAction(pod, inplaceUpdateSidecarNames); err != nil {
 		return reconcile.Result{}, err
 	}
-
+	if err = r.executeKillContainerAction(pod, normalSidecarNames); err != nil {
+		return reconcile.Result{}, err
+	}
+	if ignoreExitCodeSidecarNames.Len() == 0 || !containersCompleted(pod, normalSidecarNames) || !containersCompleted(pod, inplaceUpdateSidecarNames) {
+		return reconcile.Result{}, nil
+	}
```

Comment:
plz leave a comment to explain why ignore exit code is not the default behavior and what is the potential risk of enable ingore exit code.